core/src/rpc.rs

```diff
@@ -413,6 +413,30 @@ impl JsonRpcRequestProcessor {
             .ok()
             .unwrap_or(None))
     }
+
+    pub fn get_transaction_status_history(
+        &self,
+        signature: Signature,
+    ) -> RpcResponse<Option<RpcTransactionStatus>> {
+        let root = self.bank_forks.read().unwrap().root();
+        self.blockstore
+            .get_transaction_status(signature, root)
+            .map_err(|_| Error::internal_error())
+            .map(|result| {
+                let mut context = RpcResponseContext { slot: root };
+                let value = result.and_then(|(slot, status)| {
+                    // If a slot is newer than the root in bank_forks, but not present in
+                    // bank_forks, it is a dead fork that will never be rooted
+                    if slot > root && !self.bank_forks.read().unwrap().banks.contains_key(&slot) {
```

Comment:
So this means that it's not possible for the blockstore to have record of a slot that isn't tracked yet in bank forks?

Reply:
I think this is true, but just making sure :)

Reply:
Yes, that's correct. Both of the places that we send data to be written to the transaction_status column in Blockstore (banking_stage for initial tx processing, or blockstore_processor called from replay_stage for replay) are positioned immediately after those transactions are committed to the bank.